src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java

```diff
@@ -0,0 +1,135 @@
+            18 + 18 * rowSize + 94)
+            .label(10, 5, getMetaFullName())
+            .widget(new SlotWidget(importItems, 0, 18, 18, true, true)
+                .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.IN_SLOT_OVERLAY));
+
+        for (int y = 0; y < rowSize; y++) {
+            for (int x = 0; x < rowSize; x++) {
+                int index = y * rowSize + x;
+                builder.widget(new SlotWidget(exportItems, index, 89 - rowSize * 9 + x * 18, 18 + y * 18, true, false)
+                    .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.OUT_SLOT_OVERLAY));
+            }
+        }
+
+        builder.bindPlayerInventory(entityPlayer.inventory, GuiTextures.SLOT, 8, 18 + 18 * rowSize + 12);
+        return builder.build(getHolder(), entityPlayer);
+    }
+
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote && energyContainer.getEnergyStored() >= energyAmountPerFish && getTimer() % fishingTicks == 0L &&
+            importItems.getStackInSlot(0).getCount() > 0) {
+            WorldServer world = (WorldServer) this.getWorld();
+            BlockPos waterCheckPos = getPos().down();
+            if (world.getBlockState(waterCheckPos).getBlock().getRegistryName().toString().equals("minecraft:water")) {
+                LootTable table = world.getLootTableManager().getLootTableFromLocation(new ResourceLocation("minecraft:gameplay/fishing"));
+                List<ItemStack> itemStacks = table.generateLootForPools(world.rand, new LootContext.Builder(world).build());
+                addToInventoryOrDropItems(itemStacks);
+                importItems.getStackInSlot(0).shrink(1);
+                energyContainer.removeEnergy(energyAmountPerFish);
+            }
+        }
+    }
+
+    private void addToInventoryOrDropItems(List<ItemStack> drops) {
+        EnumFacing outputFacing = getFrontFacing();
+        double itemSpawnX = getPos().getX() + 0.5 + outputFacing.getFrontOffsetX();
+        double itemSpawnY = getPos().getY() + 0.5 + outputFacing.getFrontOffsetY();
+        double itemSpawnZ = getPos().getZ() + 0.5 + outputFacing.getFrontOffsetZ();
+        for(ItemStack itemStack : drops) {
+            ItemStack remainStack = ItemHandlerHelper.insertItemStacked(exportItems, itemStack, false);
+            if(!remainStack.isEmpty()) {
+                EntityItem entityitem = new EntityItem(getWorld(), itemSpawnX, itemSpawnY, itemSpawnZ, remainStack);
+                entityitem.setDefaultPickupDelay();
+                getWorld().spawnEntity(entityitem);
+            }
+        }
+    }
+
+    @Override
+    protected IItemHandlerModifiable createImportItemHandler() {
+        return new ItemStackHandler(1) {
+            @Nonnull
+            @Override
+            public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate) {
+                if(stack.getItem().getRegistryName().toString().equals("minecraft:string")) {
```

Comment:
same here - do not use registry name for block checks, and using Ore Dictionary would be better here
`OreDictUnifier.getOreDictNames(itemStack).contains("string")`